javascript/javascript2/week3/warmup.js

```diff
@@ -0,0 +1,115 @@
+ Now create a new third function that has one parameter: planetLogFunction. 
+ The only thing the third function should do is call the provided parameter function. 
+ Try call the third function once with the Earth function and once with the Saturn function. */
+const earthLogger = function() {
+    console.log("I am from Earth, 'log from task 4'");
+}
+const saturnLogger = function() {
+    console.log("I am from Saturn, 'log from task 4'");
+}
+function planetLogFunction(chosenPlanet) {
+    chosenPlanet();
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+/* Create a button with the text called "Log location". When this button is clicked the location (latitude, longitude)
+ of the user should be logged out using this browser api */
+function geoFindMe() {
+    const status = document.getElementById("geo-status");
+    const mapLink = document.getElementById("map-link");
+    mapLink.href = "";
+    mapLink.textContent = "";
+  
+    function success(position) {
+      const latitude  = position.coords.latitude;
+      const longitude = position.coords.longitude;
+  
+      status.textContent = "";
+      mapLink.href = `https://www.openstreetmap.org/#map=18/${latitude}/${longitude}`;
+      mapLink.textContent = `This is the Latitude: ${latitude} °. This is the Longitude ${longitude} °`;
+      mapLink.style.textDecoration = "none";
+      mapLink.style.color = "black";
+    }
+  
+    function error() {
+      status.textContent = "Unable to retrieve your location";
+    }
+  
+    if(!navigator.geolocation) {
+      status.textContent = "Geolocation is not supported by your browser";
+    } else {
+      status.textContent = "Locating…";
+      navigator.geolocation.getCurrentPosition(success, error);
+    }
+  
+  }
+  
+document.getElementById("tracker").addEventListener("click", geoFindMe);
+
+// Now show that location on a map using fx the Google maps api
+// let map;
+// function initMap() {
+// map = new google.maps.Map(document.getElementById("map"), {
+//     center: { lat: -34.397, lng: 150.644 },
+//     zoom: 8,
+//     });
+// }
+// initMap();
+
+/* Create a function called runAfterDelay. It has two parameters: delay and callback. 
+When called the function should wait delay seconds and then call the provided callback function. 
+Try and call this function with different delays and different callback functions. */
+function runAfterDelay(delay, callback) {
+    console.log(`Wait for ${delay} seconds... 'log from task 7'`);
+    setTimeout(callback, delay * 1000);
+}
+runAfterDelay(3, function() {
+    console.log("Logged after 3 seconds. 'log from task 7'");
+});
+
+/* Check if we have double clicked on the page. A double click is defined by two clicks within 0.5 seconds. 
+If a double click has been detected, log out the text: "double click!" */
+document.addEventListener("dblclick", function() {
+    console.log("double click");
+});
```

Comment:
It works but try to rewrite using `click `event and `setTimeout`.

Reply:
Ok. Will change that. Thanks.